src/binary_cache.cpp

```diff
@@ -123,6 +123,8 @@ KDb GetDb(const TargetProperties& target, size_t num_cu)
     boost::filesystem::path sys_path = sys_dir / (Handle::GetDbBasename(target, num_cu) + ".kdb");
     if(user_dir.empty())
         user_path = user_dir;
+    if(!boost::filesystem::exists(sys_path))
+        sys_path = sys_dir / (target.DbId() + ".kdb");
```

Comment:
Why we initialize sys_path with GetDbBasename() and if file does not exist change sys_path to target.DbId()?
What is valid .kdb filename?

Reply:
@cderb ping for the above review opinion.

Reply:
@cderb Thanks, Chris. AFAICS this is not just an `arch` but `arch-with-encoded-target-features` (returned by target.DbId). When we open system KDB, we are looking for `arch-with-encoded-target-features_numCUs.kdb` first, then, if not found, repeat the same but without `_numCUs`.  The first variant (with numCUs) is no longer supplied, so this always fails.

🚧  I'll take care about cleanup of this code when time permits.

## FYI Format of "arch-with-encoded-target-features"

This is just for information. These formats are produced by `TargetProperties::InitDbId()`

### gfx906 and gfx908
We have different format for these two GPUs for the sake of compatibility with previous versions of databases, which was important during transition period.

```
target ID                target.DbId             Note
-----------------------------------------------------
gfx90X:xnack-            gfx90X_nosramecc        (1)
gfx90X:xnack+            gfx90X_nosramecc_xnack  (1)
gfx90X:sramecc-          gfx90X_nosramecc        (1)
gfx90X:sramecc+          gfx90X             
gfx90X:sramecc-:xnack-   gfx90X_nosramecc        (1)
gfx90X:sramecc-:xnack+   gfx90X_nosramecc_xnack  (1)
gfx90X:sramecc+:xnack-   gfx90X             
gfx90X:sramecc+:xnack+   gfx90X_xnack
-----------------------------------------------------
(1) Currently we do not support detection of SRAMECC state for
these targets and always assume sramecc+. Therefore
these target.DbId values do not appear in practice.
```

### Other targets
```
target ID                target.DbId             Note
-----------------------------------------------------
gfxXXX:xnack-            gfxXXX
gfxXXX:xnack+            gfxXXX_xnack
gfxXXX:sramecc-          gfxXXX
gfxXXX:sramecc+          gfxXXX_sramecc
gfxXXX:sramecc-:xnack-   gfxXXX
gfxXXX:sramecc-:xnack+   gfxXXX_xnack
gfxXXX:sramecc+:xnack-   gfxXXX_sramecc
gfxXXX:sramecc+:xnack+   gfxXXX_sramecc_xnack
```

Reply:
Yes, this appears to be in error.

Reply:
@cderb @JehandadKhan In fact, the state of SRAMECC is currently ignored in DB names. This is due to a W/A that we have for some bug in the old DKMS (W/A has been introduced in #832):
https://github.com/ROCm/MIOpen/blob/7b36cef67f3154dc3eff7471e335f5a7111f4b17/src/target_properties.cpp#L85-L91

As a result, KDB loading is currently working as expected.

## Current format of "arch-with-encoded-target-features" for all targets (including gfx906/908)
```
target ID                target.DbId             Note
-----------------------------------------------------
gfxXXX:xnack-            gfxXXX
gfxXXX:xnack+            gfxXXX_xnack
gfxXXX:sramecc-          gfxXXX
gfxXXX:sramecc+          gfxXXX
gfxXXX:sramecc-:xnack-   gfxXXX
gfxXXX:sramecc-:xnack+   gfxXXX_xnack
gfxXXX:sramecc+:xnack-   gfxXXX
gfxXXX:sramecc+:xnack+   gfxXXX_xnack
```

### Side note about "reported_sramecc"

This is different question, but actually we capture (possibly incorrect) "reported_sramecc" state, because we must use it when we build ASM kernels with COMGR. Otherwise correctness problems with FP16/BF16 on MI200 happened. See https://github.com/ROCm/MIOpen/issues/1257#issuecomment-972820638, bullet (3).

🟢 No immediate action is needed for now.